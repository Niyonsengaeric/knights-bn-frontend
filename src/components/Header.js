import React from 'react';
import '../assets/styles/components/header.scss';

const HeaderComponent = () => (
  <div>
    <h1 data-testid="header" className="box position__center">Barefoot Nomad</h1>
  </div>
);
export default HeaderComponent;
