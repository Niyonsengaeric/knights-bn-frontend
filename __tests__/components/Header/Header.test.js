import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ReactTestRenderer from 'react-test-renderer';
import Header from '../../../src/components/Header';

describe('Testing Header component', () => {
  it('should render with no error', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header />, div);
  });

  it('renders Header correctly', () => {
    const { getByTestId } = render(<Header />);
    expect(getByTestId('header')).toHaveTextContent('Barefoot Nomad');
  });
  it('matches the snapshot', () => {
    const renderer = ReactTestRenderer.create(<Header />);
    expect(renderer).toMatchSnapshot();
  });
});
