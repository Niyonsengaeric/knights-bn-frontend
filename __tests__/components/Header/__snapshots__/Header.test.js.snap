// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header component matches the snapshot 1`] = `
<div>
  <h1
    className="box position__center"
    data-testid="header"
  >
    Barefoot Nomad
  </h1>
</div>
`;
